blockchain/src/blockchain/Block.java

```diff
@@ -85,28 +95,50 @@ private String getHeader() {
         return this.timeStamp.toString() + this.blockVersion + this.previousHash;
     }
 
-    /** Calculates this \code{Block}'s id.
+    /**
+     * Calculates this \code{Block}'s id.
      * Returns a 64-digit long representation of this Block.
-     * @return This Block's id.
+     *
+     * @return this Block's id
      */
     private String calculateId() {
         return StringUtils.stringToHex(this.getHeader());
     }
 
-    /** Mines the hash of this Block.
-     *
+    /**
+     * Mines the hash of this Block.
      */
     public void mineHash() {
-        String hashBase = this.getHeader() + this.transactions.toString();
-        String targetPrefix = new String(new char[this.miningDifficulty]).replace("\0", " ");
-        while (this.hash.equals("")) {
-            if(targetPrefix.equals(StringUtils.stringToHex(hashBase + nonce).substring(0,
-                    this.miningDifficulty))) {
-                this.hash = StringUtils.stringToHex(hashBase + nonce);
+        String hashBase = getHeader() + transactions.toString();
+        String targetPrefix = new String(new char[miningDifficulty]).replace("\0", " ");
```

Comment:
Change replacement value to 0 

```suggestion
        String targetPrefix = new String(new char[miningDifficulty]).replace("\0", "0");
```